Allow translations to be inherited and overridden in subclasses (#1934)
ViewComponent/view_component: docs/guide/translations.md

<%= helpers.t("hello") %>
<%= I18n.t("hello") %>
```

## Inheritance

Translations are inherited from the component's parent class. Given a parent component with a translation file:

```yml
# app/components/parent_component.yml
en:
  hello: "Hello world!"
  greeting: "Cheers!"
```

> The translation is available in subclasses of `ParentComponent`, allowing translations to be used as is or overridden by the child component:

Reviewer: ```suggestion
The translation is available in subclasses of `ParentComponent`, allowing translations to be used as-is or overridden by the child component:
```

<%= helpers.t("hello") %>
<%= I18n.t("hello") %>
```

## Inheritance

> Translations are inherited from the component's parent class.
> For example, given a parent component with a translation file:

Reviewer: ```suggestion
Translations are inherited from the component's parent class. Given a parent component with a translation file:
```

```yml
# app/components/parent_component.yml
en:
  hello: "Hello world!"
  greeting: "Cheers!"
```

> Then the translation will be available in subclasses of `ParentComponent`.
> This allows translations to be used as is or overridden by the child component:

Reviewer: ```suggestion
The translation is available in subclasses of `ParentComponent`, allowing translations to be used as is or overridden by the child component:
```
